client/lib/feed-stream-store/test/index.js

```diff
@@ -294,46 +303,142 @@ describe( 'FeedPostList', function() {
 			expect( xPostedTo[ 0 ].siteURL ).to.equal( 'http://officetoday.wordpress.com' );
 		} );
 
 		it.skip( 'filters xposts with no metadata', function() {
-			posts = [ set( {}, 'meta.data.post', {
-				tags: { 'p2-xpost': {} },
-				metadata: false,
-				site_name: 'Example',
-				site_URL: 'http://example.wordpress.com'
-			} ) ];
+			posts = [
+				set( {}, 'meta.data.post', {
+					tags: { 'p2-xpost': {} },
+					metadata: false,
+					site_name: 'Example',
+					site_URL: 'http://example.wordpress.com',
+				} ),
+			];
 			filteredPosts = store.filterFollowedXPosts( posts );
 			expect( filteredPosts.length ).to.equal( 0 );
 		} );
 
 		it.skip( 'filters xposts with missing xpost metadata', function() {
-			posts = [ set( {}, 'meta.data.post', {
-				tags: { 'p2-xpost': {} },
-				metadata: {
-					0: {
-						key: 'unrelated',
-						value: 'unrelated'
-					}
-				},
-				site_name: 'Example',
-				site_URL: 'http://example.wordpress.com'
-			} ) ];
+			posts = [
+				set( {}, 'meta.data.post', {
+					tags: { 'p2-xpost': {} },
+					metadata: {
+						0: {
+							key: 'unrelated',
+							value: 'unrelated',
+						},
+					},
+					site_name: 'Example',
+					site_URL: 'http://example.wordpress.com',
+				} ),
+			];
 			filteredPosts = store.filterFollowedXPosts( posts );
 			expect( filteredPosts.length ).to.equal( 0 );
 		} );
 	} );
+
+	describe( 'conversations store', () => {
+		let conversations;
+		beforeEach( () => {
+			FeedStreamCache.clear();
+			conversations = PostListFactory( 'conversations' );
+		} );
+
+		it( 'should build an instance', () => {
+			expect( conversations ).to.be.ok;
+		} );
+
+		context( 'when holding some posts', () => {
+			beforeEach( () => {
+				conversations.receivePage( conversations.id, null, {
+					posts: [
+						{
+							site_ID: 1,
+							ID: 1,
+							date: '2016-09-15T00:00:00Z',
+							last_comment_date_gmt: '2017-01-01T00:00:00Z',
+							comments: [ { ID: 1 }, { ID: 2 }, { ID: 3 } ],
+						},
+					],
+				} );
+				expect( conversations.postKeys ).to.have.lengthOf( 1 );
+			} );
+
+			it( 'should filter out posts that it already has which have the same comments', () => {
+				const filteredPosts = conversations.filterNewPosts( [
+					{
+						site_ID: 1,
+						ID: 1,
+						last_comment_date_gmt: '2017-01-01T00:00:00Z',
+						comments: [ { ID: 1 }, { ID: 2 }, { ID: 3 } ],
+					},
+				] );
+				expect( filteredPosts ).to.have.lengthOf( 0 );
+			} );
+
+			it( 'should retain posts that it already has with new comments', () => {
+				const filteredPosts = conversations.filterNewPosts( [
+					{
+						site_ID: 1,
+						ID: 1,
+						date: '2016-09-15T00:00:00Z',
+						last_comment_date_gmt: '2017-01-01T01:00:00Z',
+						comments: [ { ID: 2 }, { ID: 3 }, { ID: 4 } ],
+					},
+				] );
+				expect( filteredPosts ).to.have.lengthOf( 1 );
+				expect( filteredPosts ).to.eql( [
+					{
+						blogId: 1,
+						comments: [ 2, 3, 4 ],
```

Comment:
oh nevermind.  I see whats happening here.  we are merely preserving it for further manipulation in `receivePage`.

Instead of a uniqBy maybe it should be some sort of comments merge.  Anyway, not a blocker for me.  we can struggle with the nitty gritty of accepting updates ux later